### PR TITLE
Schaefer dev

### PR DESCRIPTION
**Pull request by stewartboyd119**
The feature flag is slot 2 (prepeak nonlinearity)
with a flag value=5 for schaefer.

1. Updated plasticity to have additional args needed for schaefer
2. Added additional material properties

Added test coverage for OAT30, 60, 75, and 90
and compare to known non linear stress strain
responses

### CODE REVIEW

File: for/schaefer.for [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        dEp = dEp - MATMUL(MInverse4x4(J1), J0)
+      ELSE 
+        Call log%error('Schaefer material prepeak nonlinearity model only supports ndir + nshr=6 and 4 elements')
+      END IF
+
+      ! The newton raphson continually updates J0 (and as it converges J0) should
+      ! approach 0. Determine its norm (and refer to this as the residual)
+      residual = SQRT(DOT_PRODUCT(J0, J0))
+      ! once we're within a certain tolerance break the loop
+      if (residual .lt. 1e-6) THEN
```
Comment (aFrankLion) [inline for/schaefer.for]: For all convergence criteria and loop limits, please either define a named, commented variables in the subroutine, e.g.,
schaefer_nr_tolerance = 1d-6  ! convergence criterion for Newton-Raphson loop
counter_limit =1000000  ! maximum number of Newton-Raphson loops before divergence is assumed
...or create user-definable parameters and add them to the parameters module.

File: for/matProp.for [hunk 2]
```diff
@@ -24,6 +24,7 @@ Module matProp_Mod
     Double Precision :: cl                                           ! Opt. fiber nonlinearity (defaults to zero, which is no fiber nonlinearity)
     Double Precision :: mu                                               ! Opt. Friction. Defaults to zero
     Double Precision :: es(4), gs(4)
+    Double Precision :: a6, b2, n, A                                       ! Specified parameters for schaefer theory
```
Comment (aFrankLion) [inline for/matProp.for]: I'm concerned about the very generic names n and A. Consider creating more descriptive names to avoid confusion.

**Review by aFrankLion — CHANGES_REQUESTED** [review]
Great work!

Why is tests/fakemat.props added instead of updating the existing .props file?

File: for/schaefer.for (hunk 1 above)
Comment (stewartboyd119) [reply for/schaefer.for]: Created local variables per your suggestion. For simplicity, now I haven't put these in the parameters module though its probably a better solution in the long term

File: for/matProp.for (hunk 2 above)
Comment (stewartboyd119) [reply for/matProp.for]: I changed these variables (and the associated properties file) to now have schaefer_a6, schaefer_b2...

Comment (stewartboyd119) [reply for/matProp.for]: I was using fakemat.props for convenience originally. I merged the additional parameters into IM7-8552.props and updated the tests to use it as well

File: tests/IM7-8552.props [hunk 3]
```diff
@@ -123,6 +123,15 @@ gs1 = -1.47d0
 gs2 =  5.05d-1
 gs3 = -1.18d-2
 
+// Schaefer Theory (values below are from Schaefer's PPT)
```
Comment (aFrankLion) [inline tests/IM7-8552.props]: I do not think we will include Joe's PowerPoint files detailing the derivations and curve fitting in the repository. The off-axis tension and compression tests and data were published in an American Society for Composites paper, see:
https://ntrs.nasa.gov/search.jsp?R=20170010326

**Review by aFrankLion — CHANGES_REQUESTED** [review]
Code looks good. However, the readme.md needs to be updated with at least a brief description of the approach and how to apply it. Updates should include at least the following sections:
- https://github.com/nasa/CompDam_DGD/blob/master/readme.md#shear-nonlinearity
- https://github.com/nasa/CompDam_DGD/blob/master/readme.md#material-properties
- https://github.com/nasa/CompDam_DGD/blob/master/readme.md#state-variables

**Review by aFrankLion — APPROVED** [review]

**Pull request merged**